src/clarity-angular/utils/focus-trap/focus-trap.directive.ts

```diff
@@ -22,4 +37,11 @@ export class FocusTrapDirective implements AfterViewInit {
         const nativeElement: HTMLElement = this.elementRef.nativeElement;
         nativeElement.setAttribute("tabindex", "0");
     }
+
+    ngOnDestroy() {
+        if (this.focusTrapLinker.previousFocusTrap) {
+            this.focusTrapLinker.currentFocusTrap = this.focusTrapLinker.previousFocusTrap;
```

Comment:
As I was guessing above, I see 2 bugs here:
- If I destroy a `FocusTrapDirective` that is not currently the active one, you switch from the current to the previous anyway, even though the current is still there. This will break everything if the case happens.
- If I destroy a `FocusTrapDirective` and I don't have a previous one, you never unset the `currentFocusTrap` property. That's a memory leak. Even worse, that means if I open a new focus trap later, when I destroy that new one the app will try to re-enable the old one that was destroyed long ago. Once again, potential crash or broken behavior.

Reply:
Ah yes, I think you are taking about the case when `focusTrapEnabled` is not enabled, right? If so yeah, I know I have another thing to add there. The second one puzzles me a little bit. So I will discuss it with you on Slack.